lib/schema/generate.js

```diff
@@ -0,0 +1,243 @@
+                itemTypeSchema['enum'] = possibleValues;
+              }
+
+              itemTypeSchema['type'] = Array.from(new Set(possibleValues.map(v => typeof v)));
+            }
+          }
+        }
+
+        if (customErrors.length) {
+          itemTypeSchema.errorMessage = Object.fromEntries(customErrors);
+        }
+
+        if (field.isCollection) {
+          schema['type'] = 'array';
+          schema['items'] = itemTypeSchema;
+        } else {
+          schema = itemTypeSchema;
+        }
+
+        if (field.nullable) {
+          const type = [];
+          if (Array.isArray(schema.type)) {
+            type.push(...schema.type, 'null');
+          } else {
+            type.push(schema.type, 'null');
+          }
+          if (field.isCollection) {
+            if (schema.enum) {
+              schema.enum?.push(null);
+            }
+            schema.type = type;
+            if (Array.isArray(schema.items.type)) {
+              schema.items.type.push('null');
+            }
+          } else {
+            schema.items['type'] = type;
+            if (schema.items.enum) {
+              schema.items.enum?.push(null);
+            }
+          }
+        }
+      } else {
+        const fieldType = typeMappings[field.type] || 'object';
+        schema['type'] = fieldType;
+        if (field.maxLength) {
+          schema['maxLength'] = field.maxLength;
+          customErrors.push(customErrorsMapping['maxLength'](field.maxLength));
+        }
+        if (field.type === EDM_DATE_TIME_OFFSET) schema['format'] = 'date-time';
+        if (field.type === EDM_DATE) schema['format'] = 'date';
+        if (fieldType === 'integer' || fieldType === 'number') {
+          if (!field.scale && field.precision) {
+            if (field.precision) schema['maximum'] = Number('9'.repeat(field.precision));
+          }
+        }
+
+        if (!isSimpleType(field.type)) {
+          const possibleValues = getPossibleLookupValues(field.type);
+          if (possibleValues.length > 0) {
+            if (!additionalProperties) {
+              schema['enum'] = possibleValues;
+            }
+            schema['type'] = Array.from(new Set(possibleValues.map(v => typeof v)));
+          }
+        }
+
+        if (customErrors.length) {
+          schema.errorMessage = Object.fromEntries(customErrors);
+        }
+
+        if (field.nullable) {
+          const type = [];
+          if (Array.isArray(schema.type)) {
+            type.push(...schema.type, 'null');
+          } else {
+            type.push(schema.type, 'null');
+          }
+          if (schema.enum) {
+            schema.enum?.push(null);
+          }
+          schema.type = type;
+        }
+      }
+      properties[fieldName] = schema;
+    });
+
+    definitions[resourceName] = {
+      type: 'object',
+      properties,
+      additionalProperties
+    };
+  }
+
+  return definitions;
+}
+
+function createSchema(resources, lookups, additionalProperties) {
```

Comment:
Would prefer to see comments for each item in the code. 

Also, should always use arrow functions over function.